packages/flutter/test/widgets/slivers_test.dart

```diff
@@ -308,6 +308,44 @@ void main() {
     },
   );
 
+  testWidgets(
+    'SliverGrid negative usableCrossAxisExtent',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: Center(
+            child: SizedBox(
+              width: 4,
+              height: 4,
+              child: CustomScrollView(
+                slivers: <Widget>[
+                  SliverGrid(
+                    gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+                      crossAxisCount: 2,
+                      crossAxisSpacing: 8,
+                      mainAxisSpacing: 8,
+                    ),
+                    delegate: SliverChildListDelegate(
+                      <Widget>[
+                        Container(child: const Center(child: Text('A'))),
+                        Container(child: const Center(child: Text('B'))),
+                        Container(child: const Center(child: Text('C'))),
+                        Container(child: const Center(child: Text('D'))),
+                      ],
+                    ),
+                  ),
+                ],
+              ),
+            ),
+          ),
+        ),
+      );
+
+      expect(tester.takeException(), isNull);
+    },
+  );
+
```

Comment:
does this fail when you don't include the change to sliver_grid.dart?

Reply:
@Hixie Yes, it will throw rendering exception in master branch:

```
Running "flutter pub get" in flutter...                             1.8s
══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
The following assertion was thrown during performLayout():
'package:flutter/src/rendering/sliver_grid.dart': Failed assertion: line 163 pos 15:
'childMainAxisExtent != null && childMainAxisExtent >= 0': is not true.

Either the assertion indicates an error in the framework itself, or we should provide substantially
more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=BUG.md

When the exception was thrown, this was the stack:
#2      new SliverGridRegularTileLayout (package:flutter/src/rendering/sliver_grid.dart:163:15)
#3      SliverGridDelegateWithFixedCrossAxisCount.getLayout (package:flutter/src/rendering/sliver_grid.dart:335:12)
#4      RenderSliverGrid.performLayout (package:flutter/src/rendering/sliver_grid.dart:529:51)
#5      RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#6      RenderViewportBase.layoutChildSequence (package:flutter/src/rendering/viewport.dart:486:13)
#7      RenderViewport._attemptLayout (package:flutter/src/rendering/viewport.dart:1491:12)
#8      RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1400:20)
#9      RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#10     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#11     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#12     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#13     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#14     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#15     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#16     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#17     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#18     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#19     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#20     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#21     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#22     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#23     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#24     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#25     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#26     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#27     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#28     RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:270:13)
#29     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#30     RenderPositionedBox.performLayout (package:flutter/src/rendering/shifted_box.dart:396:13)
#31     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#32     RenderView.performLayout (package:flutter/src/rendering/view.dart:169:13)
#33     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1645:7)
#34     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:896:18)
#35     AutomatedTestWidgetsFlutterBinding.drawFrame (package:flutter_test/src/binding.dart:1091:23)
#36     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:298:5)
#37     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1117:15)
#38     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1055:9)
#39     AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:961:9)
#42     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#43     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:948:27)
#44     WidgetTester.pumpWidget.<anonymous closure> (package:flutter_test/src/widget_tester.dart:524:22)
#47     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#48     WidgetTester.pumpWidget (package:flutter_test/src/widget_tester.dart:521:27)
#49     main.<anonymous closure> (file:///Users/gaoge/Develop/flutter/packages/flutter/test/widgets/slivers_test.dart:314:20)
#50     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:146:29)
<asynchronous suspension>
#51     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart)
#52     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:784:19)
<asynchronous suspension>
#55     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:764:14)
#56     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1173:24)
#57     FakeAsync.run.<anonymous closure>.<anonymous closure> (package:fake_async/fake_async.dart:178:54)
#62     withClock (package:clock/src/default.dart:48:10)
#63     FakeAsync.run.<anonymous closure> (package:fake_async/fake_async.dart:178:22)
#68     FakeAsync.run (package:fake_async/fake_async.dart:178:7)
#69     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1170:15)
#70     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:138:24)
#71     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:175:19)
<asynchronous suspension>
#72     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#77     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:173:13)
#78     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:231:15)
#83     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:228:5)
#84     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:383:17)
<asynchronous suspension>
#85     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart)
#90     Invoker._onRun.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:370:9)
#91     Invoker._guardIfGuarded (package:test_api/src/backend/invoker.dart:415:15)
#92     Invoker._onRun.<anonymous closure> (package:test_api/src/backend/invoker.dart:369:7)
#99     Invoker._onRun (package:test_api/src/backend/invoker.dart:368:11)
#100    LiveTestController.run (package:test_api/src/backend/live_test_controller.dart:153:11)
#101    RemoteListener._runLiveTest.<anonymous closure> (package:test_api/src/remote_listener.dart:256:16)
#106    RemoteListener._runLiveTest (package:test_api/src/remote_listener.dart:255:5)
#107    RemoteListener._serializeTest.<anonymous closure> (package:test_api/src/remote_listener.dart:208:7)
#125    _GuaranteeSink.add (package:stream_channel/src/guarantee_channel.dart:125:12)
#126    new _MultiChannel.<anonymous closure> (package:stream_channel/src/multi_channel.dart:159:31)
#130    CastStreamSubscription._onData (dart:_internal/async_cast.dart:85:11)
#164    new _WebSocketImpl._fromSocket.<anonymous closure> (dart:_http/websocket_impl.dart:1145:21)
#172    _WebSocketProtocolTransformer._messageFrameEnd (dart:_http/websocket_impl.dart:338:23)
#173    _WebSocketProtocolTransformer.add (dart:_http/websocket_impl.dart:232:46)
#183    _Socket._onData (dart:io-patch/socket_patch.dart:2044:41)
#192    new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1580:33)
#193    _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:1076:14)
(elided 117 frames from class _AssertionError, dart:async, and package:stack_trace)

The following RenderObject was being processed when the exception was fired: RenderSliverGrid#5eb0d relayoutBoundary=up1 NEEDS-PAINT:
  creator: SliverGrid ← Viewport ← IgnorePointer-[GlobalKey#b5d38] ← Semantics ← _PointerListener ←
    Listener ← _GestureSemantics ←
    RawGestureDetector-[LabeledGlobalKey<RawGestureDetectorState>#35828] ← _PointerListener ← Listener
    ← _ScrollableScope ← _ScrollSemantics-[GlobalKey#1b0e4] ← ⋯
  parentData: paintOffset=Offset(0.0, 0.0) (can use size)
  constraints: SliverConstraints(AxisDirection.down, GrowthDirection.forward, ScrollDirection.idle,
    scrollOffset: 0.0, remainingPaintExtent: 4.0, crossAxisExtent: 4.0, crossAxisDirection:
    AxisDirection.right, viewportMainAxisExtent: 4.0, remainingCacheExtent: 254.0, cacheOrigin: 0.0)
  geometry: null
  no children current live
This RenderObject has no descendants.
════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown during performLayout():
The method 'debugAssertIsValid' was called on null.
Receiver: null
Tried calling: debugAssertIsValid()

When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      RenderViewportBase.layoutChildSequence (package:flutter/src/rendering/viewport.dart:502:34)
#2      RenderViewport._attemptLayout (package:flutter/src/rendering/viewport.dart:1491:12)
#3      RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1400:20)
#4      RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#5      RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#6      RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#7      RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#8      RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#9      RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#10     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#11     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#12     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#13     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#14     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#15     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#16     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#17     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#18     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#19     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#20     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#21     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:115:13)
#22     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#23     RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:270:13)
#24     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#25     RenderPositionedBox.performLayout (package:flutter/src/rendering/shifted_box.dart:396:13)
#26     RenderObject.layout (package:flutter/src/rendering/object.dart:1788:7)
#27     RenderView.performLayout (package:flutter/src/rendering/view.dart:169:13)
#28     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1645:7)
#29     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:896:18)
#30     AutomatedTestWidgetsFlutterBinding.drawFrame (package:flutter_test/src/binding.dart:1091:23)
#31     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:298:5)
#32     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1117:15)
#33     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1055:9)
#34     AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:961:9)
#37     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#38     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:948:27)
#39     WidgetTester.pumpWidget.<anonymous closure> (package:flutter_test/src/widget_tester.dart:524:22)
#42     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#43     WidgetTester.pumpWidget (package:flutter_test/src/widget_tester.dart:521:27)
#44     main.<anonymous closure> (file:///Users/gaoge/Develop/flutter/packages/flutter/test/widgets/slivers_test.dart:314:20)
#45     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:146:29)
<asynchronous suspension>
#46     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart)
#47     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:784:19)
<asynchronous suspension>
#50     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:764:14)
#51     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1173:24)
#52     FakeAsync.run.<anonymous closure>.<anonymous closure> (package:fake_async/fake_async.dart:178:54)
#57     withClock (package:clock/src/default.dart:48:10)
#58     FakeAsync.run.<anonymous closure> (package:fake_async/fake_async.dart:178:22)
#63     FakeAsync.run (package:fake_async/fake_async.dart:178:7)
#64     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1170:15)
#65     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:138:24)
#66     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:175:19)
<asynchronous suspension>
#67     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#72     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:173:13)
#73     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:231:15)
#78     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:228:5)
#79     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:383:17)
<asynchronous suspension>
#80     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart)
#85     Invoker._onRun.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:370:9)
#86     Invoker._guardIfGuarded (package:test_api/src/backend/invoker.dart:415:15)
#87     Invoker._onRun.<anonymous closure> (package:test_api/src/backend/invoker.dart:369:7)
#94     Invoker._onRun (package:test_api/src/backend/invoker.dart:368:11)
#95     LiveTestController.run (package:test_api/src/backend/live_test_controller.dart:153:11)
#96     RemoteListener._runLiveTest.<anonymous closure> (package:test_api/src/remote_listener.dart:256:16)
#101    RemoteListener._runLiveTest (package:test_api/src/remote_listener.dart:255:5)
#102    RemoteListener._serializeTest.<anonymous closure> (package:test_api/src/remote_listener.dart:208:7)
#120    _GuaranteeSink.add (package:stream_channel/src/guarantee_channel.dart:125:12)
#121    new _MultiChannel.<anonymous closure> (package:stream_channel/src/multi_channel.dart:159:31)
#125    CastStreamSubscription._onData (dart:_internal/async_cast.dart:85:11)
#159    new _WebSocketImpl._fromSocket.<anonymous closure> (dart:_http/websocket_impl.dart:1145:21)
#167    _WebSocketProtocolTransformer._messageFrameEnd (dart:_http/websocket_impl.dart:338:23)
#168    _WebSocketProtocolTransformer.add (dart:_http/websocket_impl.dart:232:46)
#178    _Socket._onData (dart:io-patch/socket_patch.dart:2044:41)
#187    new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1580:33)
#188    _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:1076:14)
(elided 115 frames from dart:async and package:stack_trace)

The following RenderObject was being processed when the exception was fired: RenderViewport#ac634 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE:
  needs compositing
  creator: Viewport ← IgnorePointer-[GlobalKey#b5d38] ← Semantics ← _PointerListener ← Listener ←
    _GestureSemantics ← RawGestureDetector-[LabeledGlobalKey<RawGestureDetectorState>#35828] ←
    _PointerListener ← Listener ← _ScrollableScope ← _ScrollSemantics-[GlobalKey#1b0e4] ←
    RepaintBoundary ← ⋯
  parentData: <none> (can use size)
  constraints: BoxConstraints(w=4.0, h=4.0)
  size: Size(4.0, 4.0)
  axisDirection: down
  crossAxisDirection: right
  offset: ScrollPositionWithSingleContext#2038a(offset: 0.0, range: null..null, viewport: 4.0,
    ScrollableState, AlwaysScrollableScrollPhysics -> ClampingScrollPhysics ->
    RangeMaintainingScrollPhysics, IdleScrollActivity#1bdef, ScrollDirection.idle)
  anchor: 0.0
This RenderObject had the following child:
    center child: RenderSliverGrid#5eb0d relayoutBoundary=up1 NEEDS-PAINT
════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown during paint():
The getter 'visible' was called on null.
Receiver: null
Tried calling: visible

When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      RenderViewportBase._paintContents (package:flutter/src/rendering/viewport.dart:620:26)
#2      RenderViewportBase.paint (package:flutter/src/rendering/viewport.dart:614:7)
#3      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#4      PaintingContext._repaintCompositedChild (package:flutter/src/rendering/object.dart:142:11)
#5      PaintingContext.repaintCompositedChild (package:flutter/src/rendering/object.dart:102:5)
#6      PaintingContext._compositeChild (package:flutter/src/rendering/object.dart:208:7)
#7      PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:7)
#8      RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:133:15)
#9      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#10     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:191:13)
#11     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:133:15)
#12     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#13     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:191:13)
#14     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:133:15)
#15     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#16     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:191:13)
#17     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:133:15)
#18     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#19     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:191:13)
#20     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:133:15)
#21     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#22     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:191:13)
#23     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:133:15)
#24     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#25     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:191:13)
#26     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:133:15)
#27     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#28     PaintingContext._repaintCompositedChild (package:flutter/src/rendering/object.dart:142:11)
#29     PaintingContext.repaintCompositedChild (package:flutter/src/rendering/object.dart:102:5)
#30     PaintingContext._compositeChild (package:flutter/src/rendering/object.dart:208:7)
#31     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:7)
#32     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:133:15)
#33     RenderCustomPaint.paint (package:flutter/src/rendering/custom_paint.dart:578:11)
#34     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#35     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:191:13)
#36     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:133:15)
#37     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#38     PaintingContext._repaintCompositedChild (package:flutter/src/rendering/object.dart:142:11)
#39     PaintingContext.repaintCompositedChild (package:flutter/src/rendering/object.dart:102:5)
#40     PaintingContext._compositeChild (package:flutter/src/rendering/object.dart:208:7)
#41     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:189:7)
#42     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:133:15)
#43     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#44     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:191:13)
#45     RenderShiftedBox.paint (package:flutter/src/rendering/shifted_box.dart:72:15)
#46     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#47     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:191:13)
#48     RenderView.paint (package:flutter/src/rendering/view.dart:215:15)
#49     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2323:7)
#50     PaintingContext._repaintCompositedChild (package:flutter/src/rendering/object.dart:142:11)
#51     PaintingContext.repaintCompositedChild (package:flutter/src/rendering/object.dart:102:5)
#52     PipelineOwner.flushPaint (package:flutter/src/rendering/object.dart:989:29)
#53     AutomatedTestWidgetsFlutterBinding.drawFrame (package:flutter_test/src/binding.dart:1095:27)
#54     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:298:5)
#55     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1117:15)
#56     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1055:9)
#57     AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:961:9)
#60     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#61     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:948:27)
#62     WidgetTester.pumpWidget.<anonymous closure> (package:flutter_test/src/widget_tester.dart:524:22)
#65     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#66     WidgetTester.pumpWidget (package:flutter_test/src/widget_tester.dart:521:27)
#67     main.<anonymous closure> (file:///Users/gaoge/Develop/flutter/packages/flutter/test/widgets/slivers_test.dart:314:20)
#68     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:146:29)
<asynchronous suspension>
#69     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart)
#70     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:784:19)
<asynchronous suspension>
#73     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:764:14)
#74     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1173:24)
#75     FakeAsync.run.<anonymous closure>.<anonymous closure> (package:fake_async/fake_async.dart:178:54)
#80     withClock (package:clock/src/default.dart:48:10)
#81     FakeAsync.run.<anonymous closure> (package:fake_async/fake_async.dart:178:22)
#86     FakeAsync.run (package:fake_async/fake_async.dart:178:7)
#87     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1170:15)
#88     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:138:24)
#89     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:175:19)
<asynchronous suspension>
#90     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#95     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:173:13)
#96     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:231:15)
#101    Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:228:5)
#102    Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:383:17)
<asynchronous suspension>
#103    Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart)
#108    Invoker._onRun.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:370:9)
#109    Invoker._guardIfGuarded (package:test_api/src/backend/invoker.dart:415:15)
#110    Invoker._onRun.<anonymous closure> (package:test_api/src/backend/invoker.dart:369:7)
#117    Invoker._onRun (package:test_api/src/backend/invoker.dart:368:11)
#118    LiveTestController.run (package:test_api/src/backend/live_test_controller.dart:153:11)
#119    RemoteListener._runLiveTest.<anonymous closure> (package:test_api/src/remote_listener.dart:256:16)
#124    RemoteListener._runLiveTest (package:test_api/src/remote_listener.dart:255:5)
#125    RemoteListener._serializeTest.<anonymous closure> (package:test_api/src/remote_listener.dart:208:7)
#143    _GuaranteeSink.add (package:stream_channel/src/guarantee_channel.dart:125:12)
#144    new _MultiChannel.<anonymous closure> (package:stream_channel/src/multi_channel.dart:159:31)
#148    CastStreamSubscription._onData (dart:_internal/async_cast.dart:85:11)
#182    new _WebSocketImpl._fromSocket.<anonymous closure> (dart:_http/websocket_impl.dart:1145:21)
#190    _WebSocketProtocolTransformer._messageFrameEnd (dart:_http/websocket_impl.dart:338:23)
#191    _WebSocketProtocolTransformer.add (dart:_http/websocket_impl.dart:232:46)
#201    _Socket._onData (dart:io-patch/socket_patch.dart:2044:41)
#210    new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1580:33)
#211    _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:1076:14)
(elided 115 frames from dart:async and package:stack_trace)

The following RenderObject was being processed when the exception was fired: RenderViewport#ac634:
  needs compositing
  creator: Viewport ← IgnorePointer-[GlobalKey#b5d38] ← Semantics ← _PointerListener ← Listener ←
    _GestureSemantics ← RawGestureDetector-[LabeledGlobalKey<RawGestureDetectorState>#35828] ←
    _PointerListener ← Listener ← _ScrollableScope ← _ScrollSemantics-[GlobalKey#1b0e4] ←
    RepaintBoundary ← ⋯
  parentData: <none> (can use size)
  constraints: BoxConstraints(w=4.0, h=4.0)
  layer: OffsetLayer#114a6 DETACHED
  size: Size(4.0, 4.0)
  axisDirection: down
  crossAxisDirection: right
  offset: ScrollPositionWithSingleContext#2038a(offset: 0.0, range: null..null, viewport: 4.0,
    ScrollableState, AlwaysScrollableScrollPhysics -> ClampingScrollPhysics ->
    RangeMaintainingScrollPhysics, IdleScrollActivity#1bdef, ScrollDirection.idle)
  anchor: 0.0
This RenderObject had the following child:
    center child: RenderSliverGrid#5eb0d relayoutBoundary=up1 NEEDS-PAINT
════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown during a scheduler callback:
The getter 'visible' was called on null.
Receiver: null
Tried calling: visible

When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      RenderViewportBase.visitChildrenForSemantics.<anonymous closure> (package:flutter/src/rendering/viewport.dart:216:57)
#2      WhereIterator.moveNext (dart:_internal/iterable.dart:442:13)
#3      Iterable.forEach (dart:core/iterable.dart:283:23)
#4      RenderViewportBase.visitChildrenForSemantics (package:flutter/src/rendering/viewport.dart:217:10)
#5      RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#6      RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#7      RenderIgnorePointer.visitChildrenForSemantics (package:flutter/src/rendering/proxy_box.dart:3121:14)
#8      RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#9      RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#10     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#11     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#12     RenderSemanticsAnnotations.visitChildrenForSemantics (package:flutter/src/rendering/proxy_box.dart:4450:11)
#13     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#14     RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#15     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#16     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#17     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#18     RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#19     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#20     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#21     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#22     RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#23     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#24     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#25     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#26     RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#27     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#28     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#29     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#30     RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#31     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#32     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#33     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#34     RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#35     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#36     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#37     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#38     RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#39     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#40     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#41     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#42     RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#43     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#44     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#45     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#46     RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#47     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#48     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#49     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#50     RenderObject._getSemanticsForParent.<anonymous closure> (package:flutter/src/rendering/object.dart:2678:61)
#51     RenderObjectWithChildMixin.visitChildren (package:flutter/src/rendering/object.dart:3049:14)
#52     RenderObject.visitChildrenForSemantics (package:flutter/src/rendering/object.dart:2761:5)
#53     RenderObject._getSemanticsForParent (package:flutter/src/rendering/object.dart:2673:5)
#54     RenderObject._updateSemantics (package:flutter/src/rendering/object.dart:2637:41)
#55     PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1092:16)
#56     AutomatedTestWidgetsFlutterBinding.drawFrame (package:flutter_test/src/binding.dart:1100:31)
#57     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:298:5)
#58     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1117:15)
#59     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1055:9)
#60     AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:961:9)
#63     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#64     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:948:27)
#65     WidgetTester.pumpWidget.<anonymous closure> (package:flutter_test/src/widget_tester.dart:524:22)
#68     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:72:41)
#69     WidgetTester.pumpWidget (package:flutter_test/src/widget_tester.dart:521:27)
#70     main.<anonymous closure> (file:///Users/gaoge/Develop/flutter/packages/flutter/test/widgets/slivers_test.dart:314:20)
#71     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:146:29)
<asynchronous suspension>
#72     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart)
#73     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:784:19)
<asynchronous suspension>
#76     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:764:14)
#77     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1173:24)
#78     FakeAsync.run.<anonymous closure>.<anonymous closure> (package:fake_async/fake_async.dart:178:54)
#83     withClock (package:clock/src/default.dart:48:10)
#84     FakeAsync.run.<anonymous closure> (package:fake_async/fake_async.dart:178:22)
#89     FakeAsync.run (package:fake_async/fake_async.dart:178:7)
#90     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1170:15)
#91     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:138:24)
#92     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:175:19)
<asynchronous suspension>
#93     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#98     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:173:13)
#99     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:231:15)
#104    Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:228:5)
#105    Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:383:17)
<asynchronous suspension>
#106    Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart)
#111    Invoker._onRun.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:370:9)
#112    Invoker._guardIfGuarded (package:test_api/src/backend/invoker.dart:415:15)
#113    Invoker._onRun.<anonymous closure> (package:test_api/src/backend/invoker.dart:369:7)
#120    Invoker._onRun (package:test_api/src/backend/invoker.dart:368:11)
#121    LiveTestController.run (package:test_api/src/backend/live_test_controller.dart:153:11)
#122    RemoteListener._runLiveTest.<anonymous closure> (package:test_api/src/remote_listener.dart:256:16)
#127    RemoteListener._runLiveTest (package:test_api/src/remote_listener.dart:255:5)
#128    RemoteListener._serializeTest.<anonymous closure> (package:test_api/src/remote_listener.dart:208:7)
#146    _GuaranteeSink.add (package:stream_channel/src/guarantee_channel.dart:125:12)
#147    new _MultiChannel.<anonymous closure> (package:stream_channel/src/multi_channel.dart:159:31)
#151    CastStreamSubscription._onData (dart:_internal/async_cast.dart:85:11)
#185    new _WebSocketImpl._fromSocket.<anonymous closure> (dart:_http/websocket_impl.dart:1145:21)
#193    _WebSocketProtocolTransformer._messageFrameEnd (dart:_http/websocket_impl.dart:338:23)
#194    _WebSocketProtocolTransformer.add (dart:_http/websocket_impl.dart:232:46)
#204    _Socket._onData (dart:io-patch/socket_patch.dart:2044:41)
#213    new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1580:33)
#214    _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:1076:14)
(elided 115 frames from dart:async and package:stack_trace)
════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure object was thrown running a test:
  Expected: null
  Actual: 'Multiple exceptions (4) were detected during the running of the current test, and at
least one was unexpected.'

When the exception was thrown, this was the stack:
#4      main.<anonymous closure> (file:///Users/gaoge/Develop/flutter/packages/flutter/test/widgets/slivers_test.dart:345:7)
<asynchronous suspension>
#5      main.<anonymous closure> (file:///Users/gaoge/Develop/flutter/packages/flutter/test/widgets/slivers_test.dart)
#6      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:146:29)
<asynchronous suspension>
#7      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart)
#8      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:784:19)
<asynchronous suspension>
#11     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:764:14)
#12     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1173:24)
#13     FakeAsync.run.<anonymous closure>.<anonymous closure> (package:fake_async/fake_async.dart:178:54)
#18     withClock (package:clock/src/default.dart:48:10)
#19     FakeAsync.run.<anonymous closure> (package:fake_async/fake_async.dart:178:22)
#24     FakeAsync.run (package:fake_async/fake_async.dart:178:7)
#25     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1170:15)
#26     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:138:24)
#27     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:175:19)
<asynchronous suspension>
#28     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#33     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:173:13)
#34     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:231:15)
#39     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:228:5)
#40     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:383:17)
<asynchronous suspension>
#41     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart)
#46     Invoker._onRun.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:370:9)
#47     Invoker._guardIfGuarded (package:test_api/src/backend/invoker.dart:415:15)
#48     Invoker._onRun.<anonymous closure> (package:test_api/src/backend/invoker.dart:369:7)
#55     Invoker._onRun (package:test_api/src/backend/invoker.dart:368:11)
#56     LiveTestController.run (package:test_api/src/backend/live_test_controller.dart:153:11)
#57     RemoteListener._runLiveTest.<anonymous closure> (package:test_api/src/remote_listener.dart:256:16)
#62     RemoteListener._runLiveTest (package:test_api/src/remote_listener.dart:255:5)
#63     RemoteListener._serializeTest.<anonymous closure> (package:test_api/src/remote_listener.dart:208:7)
#81     _GuaranteeSink.add (package:stream_channel/src/guarantee_channel.dart:125:12)
#82     new _MultiChannel.<anonymous closure> (package:stream_channel/src/multi_channel.dart:159:31)
#86     CastStreamSubscription._onData (dart:_internal/async_cast.dart:85:11)
#120    new _WebSocketImpl._fromSocket.<anonymous closure> (dart:_http/websocket_impl.dart:1145:21)
#128    _WebSocketProtocolTransformer._messageFrameEnd (dart:_http/websocket_impl.dart:338:23)
#129    _WebSocketProtocolTransformer.add (dart:_http/websocket_impl.dart:232:46)
#139    _Socket._onData (dart:io-patch/socket_patch.dart:2044:41)
#148    new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1580:33)
#149    _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:1076:14)
(elided 111 frames from dart:async and package:stack_trace)

This was caught by the test expectation on the following line:
  file:///Users/gaoge/Develop/flutter/packages/flutter/test/widgets/slivers_test.dart line 345
The test description was:
  SliverGrid negative usableCrossAxisExtent
════════════════════════════════════════════════════════════════════════════════════════════════════
Test failed. See exception logs above.
The test description was: SliverGrid negative usableCrossAxisExtent

✖ SliverGrid negative usableCrossAxisExtent
Exited (1)
```